### ISSUE TITLE
Wrong error output ("mismatching dependencies") when nightly is used

### ISSUE THREAD

**Issue by cmichi** [paritytech/cargo-contract#746]
```bash
$ rustup show active-toolchain nightly-aarch64-apple-darwin
$ cargo contract build                              
ERROR: Mismatching versions of `parity-scale-codec` were found!
Please ensure that your contract and your ink! dependencies use a compatible version of this package.
```

This error output is incorrect, the dependencies are all compatible.

The error happens because the function `assert_compatible_ink_dependencies()` executes `util::invoke_cargo(…)`, which in turn executes `assert_channel()?;`. The `assert_channel()` function fails because the toolchain is `nightly`. The error is bubbled up and the function `assert_compatible_ink_dependencies` erroneously assumes it was because of incompatible dependencies.

The error handling has to be improved here, so that the correct output is printed ("Contracts should be built using stable").

**Comment by boyswan** [comment]
I'm coming across this same error messages when _not_ using nightly:

```
❯ rustup default
stable-x86_64-apple-darwin (default)

❯ cargo contract -V
cargo-contract 2.0.0-alpha.3-unknown-x86_64-apple-darwin
```
```
[dependencies]
ink = "4.0.0-alpha.3"
```

If I downgrade cargo-contract to 2.0.0-alpha.2, then I get an `ERROR: No 'ink_lang' dependency found`.  I've also tried using `master` from the ink repo, but keep running into one of these two errors.

**Comment by HCastano** [comment]
Closed via #848
